### PR TITLE
[RG-161] Extend Help menu with 3 actions. Implemented scrollable licences viewer

### PR DESCRIPTION
**Pull request by TarasSydor**
> ### Motivate of the pull request
Given PR implements RG-161 feature: adds 3 extra actions to Help menu and scrollable viewer for licenses,

> ### Describe the technical details
Documentation and Relese Notes actions aren't implemented yet.
Licenses opens QPlainTextEdit based read-only viewer, containing application specific licenses. Those are supposed to be stored at share/appName/licenses.
It reads all txt files out of that folder and adds the to the view. File contents are split by 3 empty lines separator.
In case such folder doesn't exist (in FOEDAG), Licenses action does nothing,
>
> #### What does this pull request change?

image
image

License viewer takes 2/3 part of the screen

> ### Which part of the code base require a change
> 
> - [+] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [+] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -1206,3 +1226,12 @@ void MainWindow::onShowStopMessage(bool showStopCompilationMsg) {
...
   m_settings.setValue(SHOW_STOP_COMPILATION_MESSAGE_KEY, m_askStopCompilation);
 }
+
+void MainWindow::onShowLicenses() {
+  auto licensePath = GlobalSession->Context()->DataPath() / LICENSES_DIR;
+  if (!FileUtils::FileExists(licensePath)) return;
+
+  LicenseViewer viewer(licensePath, m_projectInfo.name, this);
+  centerWidget(viewer);
+  viewer.exec();
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:1236]: Qt documents says:
Note: Avoid using this function; instead, use open(). Unlike exec(), open() is asynchronous, and does not spin an additional event loop. This prevents a series of dangerous bugs from happening (e.g. deleting the dialog's parent while the dialog is open via exec()). When using open() you can connect to the finished() signal of QDialog to be notified when the dialog is closed.

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:1236]: but maybe it is fine in this case

**Comment by TarasSydor** [comment]
> @TarasSydor please provide screenshot with whole dialog but not a part.

that was entire dialog, maybe just without a title :)
added a screenshot of entire application

File: src/MainWindow/main_window.cpp (hunk 1 above)
Comment (TarasSydor) [reply src/MainWindow/main_window.cpp:1236]: for some reason neither show() nor open() show the dialog, so I prefer keeping it as is.
About dialog is also shown via exec().

**Review by skyler-rs — APPROVED** [review]

**Comment by KochynVolodymyr** [comment]
@ravic-rs I think for the Licenses window, button Close or Ok should be available at the bottom since all window like this have some button.

**Comment by TarasSydor** [comment]
@ravic-rs 
It seems the original issue (EDA-487) got updated and there is no 'License' action in Help menu anymore:

- Release notes (I am assuming that this also includes all the known issues and bug fixes)
- User guide
- Videos and Tutorials
- Quick Help guide (if we have it available or we can have Tony’s quickstart guide here)

Do we still need to see licenses in the application?

**Comment by TarasSydor** [comment]
> @ravic-rs I think for the Licenses window, button Close or Ok should be available at the bottom since all window like this have some button. Close window by press Enter will not work without button.

close button added (+ screenshot updated)

File: src/MainWindow/main_window.cpp (hunk 1 above)
Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:1236]: @KochynVolodymyr are you still waiting for this to be resolved ?

**Pull request merged**